### PR TITLE
accomodating content insets and content container padding in visible …

### PR DESCRIPTION
**Pull request by rajatgupta26**
…row calculations

### CODE REVIEW

File: src/core/RecyclerListView.tsx [hunk 1]
```diff
@@ -107,7 +107,7 @@ export interface RecyclerListViewProps {
 
     //For all props that need to be proxied to inner/external scrollview. Put them in an object and they'll be spread
     //and passed down. For better typescript support.
-    scrollViewProps?: object;
+    scrollViewProps?: ScrollViewProperties;
```
Comment (naqvitalha) [inline src/core/RecyclerListView.tsx]: Web build will fail here

File: src/core/RecyclerListView.tsx [hunk 2]
```diff
@@ -314,6 +316,26 @@ export default class RecyclerListView extends React.Component<RecyclerListViewPr
         //     rowRenderer,
         //     ...props,
         // } = this.props;
+
+        if (!this._derivedContentInsets && this.props.scrollViewProps) {
```
Comment (naqvitalha) [inline src/core/RecyclerListView.tsx]: Make a utility and move to componentWillReceiveProps.
Render is called multiple times in recyclerview internally.

File: src/core/RecyclerListView.tsx [hunk 3]
```diff
@@ -40,7 +40,7 @@ import ItemAnimator, { BaseItemAnimator } from "./ItemAnimator";
 import ScrollComponent from "../platform/reactnative/scrollcomponent/ScrollComponent";
 import ViewRenderer from "../platform/reactnative/viewrenderer/ViewRenderer";
 import { DefaultJSItemAnimator as DefaultItemAnimator } from "../platform/reactnative/itemanimators/defaultjsanimator/DefaultJSItemAnimator";
-import { Platform } from "react-native";
+import { Platform, ScrollViewProperties, Insets } from "react-native";
```
Comment (naqvitalha) [inline src/core/RecyclerListView.tsx]: Define this type yourself

Comment (rajatgupta26) [reply src/core/RecyclerListView.tsx]: Is this type also not available for web?

File: src/core/RecyclerListView.tsx [hunk 4]
```diff
@@ -314,6 +316,26 @@ export default class RecyclerListView extends React.Component<RecyclerListViewPr
         //     rowRenderer,
         //     ...props,
         // } = this.props;
+
+        if (!this._derivedContentInsets && this.props.scrollViewProps) {
+            this._derivedContentInsets = this.props.scrollViewProps.contentInset ? this.props.scrollViewProps.contentInset : { top: 0, left: 0, right: 0, bottom: 0 };
+
+            if (this.props.scrollViewProps.contentContainerStyle) {
```
Comment (naqvitalha) [inline src/core/RecyclerListView.tsx]: contentContainerStyle can also come in props

File: src/core/RecyclerListView.tsx [hunk 5]
```diff
@@ -108,6 +115,12 @@ export interface RecyclerListViewProps {
     //For all props that need to be proxied to inner/external scrollview. Put them in an object and they'll be spread
     //and passed down. For better typescript support.
     scrollViewProps?: object;
+
+    /**
+     * Use this prop to provide padding around list content instead of
+     * using padding/margin in contentContainerStyle prop of scroll view.
+     */
+    contentInsets?: EdgeInsets;
```
Comment (naqvitalha) [inline src/core/RecyclerListView.tsx]: Won't this conflict with existing ScrollView prop?

File: src/core/RecyclerListView.tsx [hunk 6]
```diff
@@ -319,6 +335,14 @@ export default class RecyclerListView extends React.Component<RecyclerListViewPr
                 ref={(scrollComponent) => this._scrollComponent = scrollComponent as BaseScrollComponent | null}
                 {...this.props}
                 {...this.props.scrollViewProps}
+                {...{
+                    contentContainerStyle: {
+                        paddingTop: this._contentInsets.top,
```
Comment (naqvitalha) [inline src/core/RecyclerListView.tsx]: All are `top`, typo

File: src/core/RecyclerListView.tsx [hunk 7]
```diff
@@ -319,6 +335,14 @@ export default class RecyclerListView extends React.Component<RecyclerListViewPr
                 ref={(scrollComponent) => this._scrollComponent = scrollComponent as BaseScrollComponent | null}
                 {...this.props}
                 {...this.props.scrollViewProps}
+                {...{
+                    contentContainerStyle: {
```
Comment (naqvitalha) [inline src/core/RecyclerListView.tsx]: Let's resolve conflicts if dev already sends this prop

File: src/core/RecyclerListView.tsx (hunk 5 above)
Comment (naqvitalha) [reply src/core/RecyclerListView.tsx]: Can we just take paddingTop etc directly which we can override in give contentContainerStyle.
Also will need to be handled in web version

**Pull request merged**